app/public/js/localSprites.js

```diff
@@ -111,7 +130,16 @@ function updateLocalSprite(localSprite) {
     }
     //rotation. Should there be an "isRotating" flag, ond only do this if it's true?
     localSprite.rotation += localSprite.rotationPerFrame;
+    //WRONG: Both bobbing and scale oscillation should have a nice sinusoid curve to them, but they're just linear right now.
+    //bobbing
+    if (localSprite.bobs) {
+        if (localSprite.y <= localSprite.originalY - localSprite.bobMaxY) localSprite.rising = true;
+        if (localSprite.y >= localSprite.originalY + localSprite.bobMaxY) localSprite.rising = false;
+        if (localSprite.rising) localSprite.y += localSprite.bobHeightPerFrame;
+        if (!localSprite.rising) localSprite.y -= localSprite.bobHeightPerFrame;
+    }
     //scale oscillator
+    //MAYBE WRONG: Scale might drift away from original, getting very large or small. Maybe need to implement a regulatory fix like the 'originalY' solution in the bobbing update area.
```

Comment:
You can use the same formulate above for bobbing for oscillating. Just record originalScale, and then add the sin or cos function relative to it.